tribits/doc/sphinx/sphinx_rst_generator.py

```diff
@@ -0,0 +1,254 @@
+import os
+import re
+from shutil import copyfile, copytree
+import subprocess
+import sys
+
+try:
+  doc_path = f"{os.sep}".join(os.path.abspath(__file__).split(os.sep)[:-2])
+  sys.path.append(doc_path)
+  project_path = f"{os.sep}".join(os.path.abspath(__file__).split(os.sep)[:-4])
+  sys.path.append(project_path)
+except Exception as e:
+  print(f"Can not add project path to system path! Exiting!\nERROR: {e}")
+  exit(1)
+
+
+def is_rst_file(file_path: str) -> bool:
+  """ Checks if file_path has .rst extension and if file_path is a file.
+  """
+  if os.path.splitext(file_path)[-1] == '.rst' and os.path.isfile(file_path):
+    return True
+  return False
+
+
+def change_paths_and_get_includes(source_file: str, src_file_path: str,
+    start_path: str, rst_dir: str, copy_file: bool = True) -> tuple:
+  """ Changes paths in source file, to be relative to sphinx_path or parent .rst
+    document.
+    Returns a tuple with .rst file content and includes(absolute_path, relative_to)
+  """
+  with open(source_file, 'r') as src_file:
+    source_file_str = src_file.read()
+    source_file_list = list()
+    include_file_list = set()
+    for line in source_file_str.split('\n'):
+      splitted_line = line.split()
+      if 'include::' in splitted_line:
+        incl_index = splitted_line.index('include::')
+        path_index = incl_index + 1
+        if len(splitted_line) > path_index:
+          new_line = []
+          spaces_indented = line.find('.') # Below 'join()' statement adds a space!
+          if spaces_indented > 0:
+            new_line.append(' '*(spaces_indented-1))
+          new_line.extend(splitted_line[:path_index])
+          abs_path = os.path.abspath(os.path.join(src_file_path,
+            splitted_line[path_index]))
+          file_name = os.path.split(abs_path)[-1]
+          new_path = os.path.join(rst_dir, file_name)
+          if not os.path.isfile(new_path) and copy_file:
+            copyfile(src=abs_path, dst=new_path, follow_symlinks=True)
+          if is_rst_file(file_path=new_path):
+            include_file_list.add(abs_path)
+          rel_path_from_sphinx_dir = os.path.relpath(path=new_path, start=start_path)
+          new_line.append(rel_path_from_sphinx_dir)
+          new_line = ' '.join(new_line)
+          source_file_list.append(new_line)
+        else:
+          source_file_list.append(line)
+      else:
+        source_file_list.append(line)
+    abs_path_str = '\n'.join(source_file_list)
+
+  return abs_path_str, include_file_list
+
+
+class SphinxRstGenerator:
+  """ Changes include paths to relative to Sphinx build dir. Saves three main .rst docs
+    files inside Sphinx dir.
+  """
+
+  def __init__(self):
+    self.paths = {
+      'maintainers_guide': {
+        'src': os.path.join(doc_path, 'guides', 'maintainers_guide',
+           'TribitsMaintainersGuide.rst'),
+        'src_path': os.path.join(doc_path, 'guides', 'maintainers_guide'),
```

Comment:
Why is this path duplicated in the 'src_path' field?  This is the base path in the of the 'src' field in every case, no?   Can't you just call `os.path.dirname()` on the `src` field and get this?

Reply:
On second thought, there are more pressing things than to refine this code.  It works.  Time to merge and move on.